OpenRA.Game/Actor.cs

```diff
@@ -118,6 +120,13 @@ internal Actor(World world, string name, TypeDictionary initDict)
 			visibilityModifiers = TraitsImplementing<IVisibilityModifier>().ToArray();
 			defaultVisibility = Trait<IDefaultVisibility>();
 			Targetables = TraitsImplementing<ITargetable>().ToArray();
+			targetablePositions = TraitsImplementing<ITargetablePositions>().ToArray();
+			this.World.AddFrameEndTask(w =>
+			{
+				// Caching this in a AddFrameEndTask, because trait construction order might cause problems if done directly at creation time
```

Comment:
Are we sure that this won't cause problems with any traits that try to query targetable positions during their own creation?

Reply:
I'm not, but I'm getting an NRE referring to HitShape line 89 when I try to do this outside an FET.